### PR TITLE
Fix DynamicModuleLoader and remove support for React-Redux 5

### PR DESCRIPTION
**Pull request by abettadapur**
This PR fixes:
* #76: We were not applying ALL of the context values that were provided from `ReactReduxContext`, so do that here.
* When using the `createStore` prop, we were not providing a modified context to the first connected component. Now, if we call `createStore` and render a `Provider`, we will render another instance of `DynamicModuleLoader`, which will actually add the modules.
* While writing this PR, I also decided to simplify the component by removing support for `react-redux@5`

This should result in a major version bump to V4

### CODE REVIEW

File: packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx [hunk 1]
```diff
@@ -118,60 +75,71 @@ class DynamicModuleLoaderImpl extends React.Component<
...
+            console.error(message);
+
+            throw new Error(message);
+        }
+
+        // Memoize the context if it has changed upstream
+        if (this.props.reactReduxContext !== this._memoizedRRContext) {
+            this._memoizedRRContext = {
+                ...this.props.reactReduxContext,
+                storeState: this.props.reactReduxContext.store.getState(),
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: Why are we memoizing state only when context changes?

File: packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx [hunk 2]
```diff
@@ -100,12 +55,14 @@ class DynamicModuleLoaderImpl extends React.Component<
     private _addedModules?: IDynamicallyAddedModule;
     private _providerInitializationNeeded: boolean = false;
     private _store: IModuleStore<any>;
-    private _getLatestState: boolean;
+    private _memoizedRRContext: any;
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: Please add some comments why we need to memoizedRRContext, it is ok to call it _memoizedReactReduxContext ... RR is not clear.

File: packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx [hunk 3]
```diff
@@ -100,12 +55,14 @@ class DynamicModuleLoaderImpl extends React.Component<
     private _addedModules?: IDynamicallyAddedModule;
     private _providerInitializationNeeded: boolean = false;
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: We should add doc comments on each member of the interface as not all are self explanatory.

**Comment by navneet-g** [comment]
Please add some unit tests ... did you test all the examples with these changes?

File: packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx (hunk 1 above)
Comment (abettadapur) [reply packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: I overwrite state now, every time on render. Not sure if that is the right approach either, but we dont want to be re-creating this object everytime

Comment (abettadapur) [reply packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: If the state changes above, then we will get a new context object here.

See: https://github.com/reduxjs/react-redux/blob/162b81a3fffe75ff6181b711777067ba6e63a34b/src/components/Provider.js#L59

this.setState({...}) returns a new object

File: packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx [hunk 4]
```diff
@@ -31,62 +26,22 @@ export interface IDynamicModuleLoaderContext {
...
-    /** Modules that need to be dynamically registerd */
-    modules: IModuleTuple;
-
-    store: IModuleStore<any>;
-
-    strictMode: boolean;
-
-    createStore?: () => IModuleStore<any>;
+interface IDynamicModuleLoaderImplProps extends IDynamicModuleLoaderProps {
+    reactReduxContext?: { store: IModuleStore<any> };
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: doc comments.

File: packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx [hunk 5]
```diff
@@ -118,60 +79,73 @@ class DynamicModuleLoaderImpl extends React.Component<
...
-        const { createStore, modules } = this.props;
-        if (!this._store) {
-            if (createStore) {
-                this._store = createStore();
-                this._providerInitializationNeeded = true;
-            } else {
-                throw new Error(
-                    "Store could not be resolved from React context"
+    public render(): React.ReactNode {
+        if (this.state.readyToRender) {
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: comments?

Comment (abettadapur) [reply packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx]: What comments are you looking for here? This seems pretty self explanatory to me

**Pull request merged**